src/cmds/testing/block_dev/block_dev_test.c

```diff
@@ -191,30 +197,57 @@ static int block_dev_test(struct block_dev *bdev, uint64_t s_block, uint64_t n_b
 		return -1;
 	}
 
-	for (uint64_t i = s_block; i < blocks; i++) {
+	for (uint64_t i = s_block; i < blocks; i += m_blocks) {
 		printf("Testing %"PRIu64" (of %"PRIu64")\n", i, total_blocks);
 
-		fill_buffer(write_buf, blk_sz);
+		int out_of_range = 0;
+		/**
+		 * When writing multiple blocks, if the block we are supposed
+		 * to write is outside the block range, then write only till
+		 * the block range and ignore the rest
+		 */
+		if (i + m_blocks >= blocks) {
+			m_blocks = blocks - i - 1;
+			out_of_range = 1;
+		}
+
+		fill_buffer(write_buf, blk_sz * m_blocks);
+
+		size_t data_sz = blk_sz * m_blocks;
+
+		/**
+		 * When writing multiple blocks, if the data we are supposed to
+		 * write, exceeds the device size, then we write till the device
+		 * size is full and ignore the rest.
+		 */
+		if(data_sz + (i * blk_sz) > bdev->size) {
+			uint64_t diff_size = (data_sz + (i * blk_sz)) - bdev->size;
```

Comment:
I am not sure about how the data could exceed the device size, because you already verified above that m_blocks suited the device size.

Reply:
Yeah, you are correct, we don't need that. I will update the commit.

Reply:
@alexkalmuk: I updated the commit with the changes. :)

Reply:
Thanks